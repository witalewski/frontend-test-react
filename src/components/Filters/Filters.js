import React, { Component } from 'react';
import c1 from '../../assets/open-now-checkbox--unchecked.png';
import c2 from '../../assets/open-now-checkbox--checked.png';

export class Filters extends Component {
  render() {
    return (
      <div className="filters">
      {/* Make webpack include the images in build */}
        <div style={{ display: 'none' }}>
          <img alt="c1" src={c1} />
          <img alt="c2" src={c2} />
        </div>
        Filter by:
        <input className="filters__open-now" type="checkbox" id="opennow" />
        <label for="opennow" />
        Open Now
      </div>
    );
  }
}

export default Filters;
