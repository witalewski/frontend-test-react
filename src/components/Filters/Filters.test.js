import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Filters } from './Filters';

describe('Filters', () => {
  it('matches snapshot', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <Filters
        filterByOpenNow
        setFilterByOpenNow={jest.fn()}
        categories={['Cafe', 'Modern European']}
        setFilterByCategory={jest.fn()}
        setFilterByPrice={jest.fn()}
      />
    );
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
