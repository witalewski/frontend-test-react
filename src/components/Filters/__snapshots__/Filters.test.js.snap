// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters matches snapshot 1`] = `
<div
  className="filters"
>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    <img
      alt="c1"
      src="open-now-checkbox--unchecked.png"
    />
    <img
      alt="c2"
      src="open-now-checkbox--checked.png"
    />
  </div>
  Filter by:
  <input
    checked={true}
    className="filters__open-now"
    id="opennow"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor="opennow"
  />
  <span
    className="filters__open-now-label"
  >
    Open Now
  </span>
  <span
    className="filters__price-label"
  >
    Price
  </span>
  <select
    onChange={[Function]}
  >
    <option
      value=""
    >
      All prices
    </option>
    <option
      value="$"
    >
      $
    </option>
    <option
      value="$$"
    >
      $$
    </option>
    <option
      value="$$$"
    >
      $$$
    </option>
    <option
      value="$$$$"
    >
      $$$$
    </option>
  </select>
  <span
    className="filters__category-label"
  >
    Category
  </span>
  <select
    onChange={[Function]}
  >
    <option
      value=""
    >
      All categories
    </option>
    <option
      value="Cafe"
    >
      Cafe
    </option>
    <option
      value="Modern European"
    >
      Modern European
    </option>
  </select>
</div>
`;
